### DISCUSSION TITLE
soundtouchjs + wavesurfer have delay or desync.

### DISCUSSION THREAD

**Discussion by tenkus47** [katspaugh/wavesurfer.js#2928] [Q&A]
i have been trying to use soundtouch js with wavesurfer in [prodigy app](https://prodi.gy/) to fix the audio pitch problem when we change the playbackrate. the thing is that it works good and now the only problem is when i click play button and if i have set the playback rate to slower , the wave start moving first and after some minisec the audio start . that causes to end audio before its done playing and the next time i click on play the remaining sound shows up.

is there a way i can sync them together.i have seen many post related to this issue but didnt really got a good answer.
- somehow if we can slowdown the speed of wave without affecting the audio rate but it doesnt seems to be a good solution.

this link gives a example of using wavesurfer with soundtouch but this shows the exact problem i am facing ,[example](https://wavesurfer-js.org/example/stretcher/) . they tried a sound that have empty sound on starting and end ,so it looks fine but my audio file directly start and end with actual sound so it is a big problem.

**Comment by katspaugh** [comment]
If your goal is just to change the speed (or speed _and_ pitch) of the audio, you don't need soundtouch, just set the desired playback rate.
See this example: https://wavesurfer-js.org/examples/#speed.js
